### ISSUE TITLE
Support for Python > 3.8

### ISSUE THREAD

**Issue by harshkhandeparkar** [metakgp/chillzone#39]
The PDF library used to read timetables, [camelot-py](https://pypi.org/project/camelot-py), only supports Python versions 3.6, 3.7, and 3.8. Support for Python 3.10+ would be mandatory in a year since 3.8 will stop receiving security updates in Oct 2024.

Possible solutions:
- Use an alternate PDF library
- Fork and update the current library

**Comment by anuraganand92** [comment]
I'm working with to use tabula-py as the alternate library to fix this and pandas to export as excel

**Comment by harshkhandeparkar** [comment]
@shikharish @ashwinpra

**Comment by shikharish** [comment]
@anuraganand92 
Go ahead!
And do share your progress. How is tabula working on the pdf?

**Comment by anuraganand92** [comment]
I discarded tabula as it wasn't that good or fast enough for parsing, I tried pdfplumber which was similar to camelot.
I attempted it on test.pdf, but i am not sure if the parsing format in test.xls is the correct one, because some cells have multiple entries or different arrangement of entries
[test.xlsx](https://github.com/metakgp/chillzone/files/12778038/test.xlsx)

**Comment by shikharish** [comment]
Yes I myself tried tabula, pdfplumber and a few others. None of them were as good as camelot.
